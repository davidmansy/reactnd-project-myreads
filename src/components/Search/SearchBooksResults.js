import React from 'react';
import BooksGrid from '../BooksGrid';
import TechnicalError from '../Common/TechnicalError';
import Loading from '../Common/Loading/Loading';

function SearchBooksResults(props) {
  const { error, isLoading, books, handleShelfChange } = props;

  return (
    <div className="search-books-results">
      {error && <TechnicalError error={error} />}
      {isLoading && <Loading />}
      {!!books.length && (
        <BooksGrid books={books} handleShelfChange={handleShelfChange} />
      )}
    </div>
  );
}

export default SearchBooksResults;
