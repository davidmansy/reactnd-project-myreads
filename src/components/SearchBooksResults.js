import React from 'react';
import BooksGrid from './BooksGrid';
import TechnicalError from './TechnicalError';
import Loading from './Loading/Loading';

function SearchBooksResults(props) {
  const { error, isLoading, books, handleShelfChange } = props;

  return (
    <div className="search-books-results">
      {error && <TechnicalError error={error} />}
      {isLoading && <Loading />}
      {!!books.length && (
        <BooksGrid books={books} handleShelfChange={handleShelfChange} />
      )}
    </div>
  );
}

export default SearchBooksResults;
