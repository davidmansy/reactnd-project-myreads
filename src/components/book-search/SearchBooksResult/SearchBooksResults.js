import React from 'react';
import BooksGrid from '../../books/BooksGrid/BooksGrid';
import TechnicalError from '../../generic/TechnicalError';
import Loading from '../../generic/Loading/index';
import PropTypes from 'prop-types';

function SearchBooksResults({ error, isLoading, books, handleShelfChange }) {
  return (
    <div className="search-books-results" data-testid="books-results">
      {error && <TechnicalError errorMessage={error} />}
      {isLoading && <Loading />}
      {!!books.length && (
        <BooksGrid books={books} handleShelfChange={handleShelfChange} />
      )}
    </div>
  );
}

SearchBooksResults.propTypes = {
  error: PropTypes.string,
  isLoading: PropTypes.bool.isRequired,
  books: PropTypes.arrayOf(PropTypes.object),
  handleShelfChange: PropTypes.func.isRequired
};

export default SearchBooksResults;
