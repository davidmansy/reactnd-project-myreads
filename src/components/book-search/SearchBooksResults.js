import React from 'react';
import BooksGrid from '../books/BooksGrid';
import TechnicalError from '../common/TechnicalError';
import Loading from '../common/Loading/index';
import PropTypes from 'prop-types';

function SearchBooksResults(props) {
  const { error, isLoading, books, handleShelfChange } = props;

  return (
    <div className="search-books-results">
      {error && <TechnicalError errorMessage={error} />}
      {isLoading && <Loading />}
      {!!books.length && (
        <BooksGrid books={books} handleShelfChange={handleShelfChange} />
      )}
    </div>
  );
}

SearchBooksResults.propTypes = {
  error: PropTypes.string,
  isLoading: PropTypes.bool.isRequired,
  books: PropTypes.arrayOf(PropTypes.object),
  handleShelfChange: PropTypes.func.isRequired
};

export default SearchBooksResults;
