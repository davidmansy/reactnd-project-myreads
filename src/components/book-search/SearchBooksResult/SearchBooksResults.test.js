import React from 'react';
import ReactDOM from 'react-dom';
import TestRenderer from 'react-test-renderer';
import SearchBooksResults from './SearchBooksResults';
import { MemoryRouter } from 'react-router-dom';
import { cleanup } from 'react-testing-library';

afterEach(cleanup);

describe('SearchBooksResults', () => {
  const props = {
    error: '',
    isLoading: false,
    books: [],
    handleShelfChange: () => {}
  };

  it('renders', () => {
    const container = document.createElement('div');
    ReactDOM.render(
      <MemoryRouter>
        <SearchBooksResults {...props} />
      </MemoryRouter>,
      container
    );
  });

  it('snapshots', () => {
    const testRenderer = TestRenderer.create(
      <MemoryRouter>
        <SearchBooksResults {...props} />
      </MemoryRouter>
    );
    let tree = testRenderer.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
