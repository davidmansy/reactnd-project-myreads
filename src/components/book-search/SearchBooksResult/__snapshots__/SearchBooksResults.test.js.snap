// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBooksResults snapshots 1`] = `
<div
  className="search-books-results"
  data-testid="books-results"
>
  
</div>
`;
